denonavr/foundation.py

```diff
@@ -622,21 +648,28 @@ async def async_options(self) -> None:
 
     async def async_settings_menu(self) -> None:
         """Options menu on receiver via HTTP get command."""
-        res = await self.api.async_get_command(self.urls.command_setup_query)
+        # fast path if we already know the state
         if self.telnet_available:
-            if res is not None and res == "MNMEN ON":
+            if self._settings_menu is None:
+                self._settings_menu = await self.api.async_get_command(
+                    self.urls.command_setup_query
+                )
+
```

Comment:
Almost 😉 

```suggestion
```
You could entirely remove this part and add a `MNMEN?` line after [this line](https://github.com/ol-iver/denonavr/blob/0e4f0338a6865759db7c9b110a2e59a1e1fb0dd9/denonavr/api.py#L569).
This command will be sent when the telnet connection has been established. The receiver will answer with either `MNMEN ON` or `MNMEN OFF` (see the api document I linked before on page 27), so `self._settings_menu` won't be None.

Reply:
Aha, I was wondering how the answers would work on Telnet :)